File: seleniumEasyJava/src/test/java/E2E/poms/dynamicData/DynamicDataPOM.java
package E2E.poms.dynamicData;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class DynamicDataPOM {
    private WebDriver webDriver;

    public DynamicDataPOM(WebDriver webDriver){
        this.webDriver = webDriver;

        PageFactory.initElements(webDriver, this);
    }

    // ------------------- Get Random User --------------------------------

    @FindBy(id = "save")
    public WebElement getNewUserButton;

    @FindBy(xpath = "//*[contains(@src, 'https://randomuser.me/')]")
    public WebElement profileImage;

    @FindBy(id = "loading")
    public WebElement name;

}

File: seleniumEasyPython/poms/dynamic_data_POM.py
from selenium.webdriver.chrome.webdriver import WebDriver
from selenium.webdriver.common.by import By
from selenium.webdriver.remote.webelement import WebElement


class DynamicDataPOM:
    def __init__(self, driver: WebDriver):
        self.driver = driver

    # ------------------- Get Random User --------------------------------

    def get_new_user(self):
        element: WebElement = self.driver.find_element(By.ID, 'save')
        return element

    def profile_image(self):
        element: WebElement = self.driver.find_element(By.XPATH, "//*[contains(@src, 'https://randomuser.me/')]")
        return element

    def first_name(self):
        element: WebElement = self.driver.find_element(By.ID, "loading")
        return element

    def last_name(self):
        element: WebElement = self.driver.find_element(By.ID, "loading")
        return element
